packages/core/src/services/scales-cartesian.ts

```diff
@@ -0,0 +1,335 @@
+						const date1 = new Date(item[domainKey]);
+						const date2 = new Date(domainValue);
+						return date1.getTime() === date2.getTime();
+					});
+
+					// assign the shared label on the data items and add them to the array
+					dataItems.forEach(item => {
+						activePoints.push(
+							Object.assign({datasetLabel: sharedLabel,
+								value: item.value,
+							}, item)
+						);
+					});
+				});
+				break;
+		}
+		return activePoints;
+	}
+
+	protected getScaleDomain(axisPosition: AxisPositions) {
+		const options = this.model.getOptions();
+		const axisOptions = Tools.getProperty(options, "axes", axisPosition);
+
+		const { datasets, labels } = this.model.getDisplayData();
+
+		// If scale is a LABELS scale, return some labels as the domain
+		if (axisOptions && axisOptions.scaleType === ScaleTypes.LABELS) {
+			if (labels) {
+				return labels;
+			} else {
+				return this.model.getDisplayData().datasets[0].data.map((d, i) => i + 1);
+			}
+		}
+
+		// Get the extent of the domain
+		let domain;
+		// If the scale is stacked
+		if (axisOptions.stacked) {
+			domain = extent(
+				labels.reduce((m, label: any, i) => {
+					const correspondingValues = datasets.map(dataset => {
+						return !isNaN(dataset.data[i]) ? dataset.data[i] : dataset.data[i].value;
+					});
+					const totalValue = correspondingValues.reduce((a, b) => a + b, 0);
+
+					// Save both the total value and the minimum
+					return m.concat(totalValue, min(correspondingValues));
+				}, [])
+					// Currently stack layouts in the library
+					// Only support positive values
+					.concat(0)
+			);
+		} else {
+			// Get all the chart's data values in a flat array
+			let allDataValues = datasets.reduce((dataValues, dataset: any) => {
+				dataset.data.forEach((datum: any) => {
+					if (axisOptions.scaleType === ScaleTypes.TIME) {
+						dataValues = dataValues.concat(datum.date);
+					} else {
+						dataValues = dataValues.concat(isNaN(datum) ? datum.value : datum);
+					}
+				});
+
+				return dataValues;
+			}, []);
+
+			if (axisOptions.scaleType !== ScaleTypes.TIME) {
+				allDataValues = allDataValues.concat(0);
+			}
+
+			domain = extent(allDataValues);
+		}
+
+		if (axisOptions.scaleType === ScaleTypes.TIME) {
+			const spaceToAddToEdges = Tools.getProperty(options, "timeScale", "addSpaceOnEdges");
+			if (spaceToAddToEdges) {
+				const startDate = new Date(domain[0]);
+				const endDate = new Date(domain[1]);
+				if (differenceInYears(endDate, startDate) > 1) {
+					return [subYears(startDate, spaceToAddToEdges), addYears(endDate, spaceToAddToEdges)];
+				}
+				if (differenceInMonths(endDate, startDate) > 1) {
+					return [subMonths(startDate, spaceToAddToEdges), addMonths(endDate, spaceToAddToEdges)];
+				}
+				if (differenceInDays(endDate, startDate) > 1) {
+					return [subDays(startDate, spaceToAddToEdges), addDays(endDate, spaceToAddToEdges)];
+				} else if (differenceInHours(endDate, startDate) > 1) {
```

Comment:
`else if` doesn't seem necessary and in combination with the `if`s above it that don't do else looks a bit peculiar.